### PR TITLE
Feature/infrastructure with custom microservices

### PR DESCRIPTION
**Pull request by adrianz-kainos**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,28 +28,27 @@ in `ApiGateBaseConfig.tf` and resources corresponding to each microservice are d
...
-* Package each microservice (run all commands from repository's root directory)
+(Optional) Set the environment variables containing a username and password of your choice. 
+If you skip this step, terraform will prompt you for username and password.
+Username "admin" is reserved by AWS.
 
-<!-- this separates the code snippet from the list element above -->
-    FOODSVC_PATH="./Microservices/FoodService"
-    mvn -f $FOODSVC_PATH/pom.xml package
+    export TF_VAR_DB_USERNAME=user
+    export TF_VAR_DB_PASSWORD=pass
```
Comment (xjedam) [inline README.md]: don't we have to set DB url as well since the configs were changed?

File: README.md [hunk 2]
```diff
@@ -22,28 +28,27 @@ in `ApiGateBaseConfig.tf` and resources corresponding to each microservice are d
...
-    export DWDEMO_PASSWORD=secret
+Build microservices with maven using `build.sh`:
+    
+    chmod +x build.sh
+    ./build.sh
 
-* Package each microservice (run all commands from repository's root directory)
+(Optional) Set the environment variables containing a username and password of your choice. 
+If you skip this step, terraform will prompt you for username and password.
+Username "admin" is reserved by AWS.
```
Comment (xjedam) [inline README.md]: Maybe lets keep this section as instructions to build and run the microservices locally and put all instructions about AWS deployment in the Infrastructure section? What do you think

Comment (xjedam) [inline README.md]: Or perhaps split this section into two: Local environemnt, and AWS - whatever makes more sense for you :)

File: Infrastructure/customer-microservice.tf [hunk 3]
```diff
@@ -1,15 +1,61 @@
...
+	provisioner "remote-exec" {
+		inline = [
+		# Upgrade java
+		"sudo yum install -y java-1.8.0",
+		"sudo yum remove -y java-1.7.0-openjdk",
+
+		# Export environment variables
+		"echo 'export DWDEMO_USER=${var.DB_USERNAME}' >> ~/.bashrc",
+		"echo 'export DWDEMO_PASSWORD=${var.DB_PASSWORD}' >> ~/.bashrc",
+		"echo 'export DWDEMO_DB=jdbc:postgresql://${aws_db_instance.default.endpoint}/microservices' >> ~/.bashrc",
```
Comment (adrianz-kainos) [inline Infrastructure/customer-microservice.tf:40]: @xjedam terraform will insert DB URL for us, after it creates the RDS instance defined in `database.tf`.

Comment (xjedam) [inline Infrastructure/customer-microservice.tf:40]: yep, meant the local scenario in readme. We won't be using terraform there

File: README.md [hunk 4]
```diff
@@ -14,6 +14,12 @@ You can define which region you want to use, and what images on which region sho
 
 ### EC2
 Two EC2 instances are created one for each microservice. Defined by `food-microservice.tf` and `customer-microservice.tf`.
+Java is upgraded to version 1.8.0 on each instance and .jar files containing respective microservices are provisioned from the local machine.
+`RunMicroservice.sh` is provisioned and launched as a cron job on the EC2 instance to ensure that the microservice is running.
+Microservice's output is written to `~/microservice.log`.
+
```
Comment (xjedam) [inline README.md:44]: Would be usefull to also mention how to ssh in to view the logs

File: README.md (hunk 1 above)
Comment (adrianz-kainos) [inline README.md]: DWDEMO_DB Added

File: README.md (hunk 4 above)
Comment (adrianz-kainos) [inline README.md:44]: Done. The ssh example uses the instance's IP provided by terraform/AWS.

File: README.md (hunk 2 above)
Comment (adrianz-kainos) [inline README.md]: Good idea. I've modified the README. Let me know what you think.

**Pull request merged**